index.js

```diff
@@ -38,14 +43,72 @@ app.use(loggerMiddleWare("dev"));
  *
  */
 
-const bodyParserMiddleWare = express.json();
-app.use(bodyParserMiddleWare);
+app.use(express.json({ limit: "50mb" }));
+app.use(express.urlencoded({ limit: "50mb", extended: true }));
+
+/**
+ *
+ * authMiddleware:
+ *
+ * When a token is provided:
+ * decrypts a jsonwebtoken to find a userId
+ * queries the database to find the user with that add id
+ * adds it to the request object
+ * user can be accessed as req.user when handling a request
+ * req.user is a sequelize User model instance
+ *
+ * When no or an invalid token is provided:
+ * returns a 4xx reponse with an error message
+ *
+ * check: auth/middleware.js
+ * for a demo check the following endpoints
+ *
+ * POST /authorized_post_request
+ * GET /me
+ *
+ */
+
+const authMiddleWare = require("./auth/middleware");
+
+/**
+ *
+ * cors middleware:
+ *
+ * Since our api is hosted on a different domain than our client
+ * we are doing "Cross Origin Resource Sharing" (cors)
+ * Cross origin resource sharing is disabled by express by default
+ * for safety reasons (should everybody be able to use your api, I don't think so!)
+ *
+ * We are configuring cors to accept all incoming requests
+ * If you want to limit this, you can look into "white listing" only certain domains
+ *
+ * docs: https://expressjs.com/en/resources/middleware/cors.html
+ *
+ */
+
+app.use((req, res, next) => {
+  res.header("Access-Control-Allow-Origin", "*");
+  next();
+});
+
+if (process.env.DELAY) {
+  app.use((req, res, next) => {
+    setTimeout(() => next(), parseInt(process.env.DELAY));
+  });
+}
 
 // Routes
 app.get("/", (req, res) => {
   res.send("Hi from express");
 });
 
+const authRouter = require("./routers/auth.js");
+
+app.use("/", authRouter);
+
+const postsRouter = require("./routers/posts.js");
```

Comment:
i know the template is coded like this, but personally i prefer all my imports/requires at the top of the file